programs/uxd/src/instructions/mint_with_router.rs

```diff
@@ -0,0 +1,437 @@
+                        collateral_amount,
+                    )?;
+                }
+                Ok(())
+            }),
+        },
+        // Credix Lp Depository 0 details
+        DepositoryInfoForMintWithRouter {
+            weight_bps: controller.credix_lp_depository_0_weight_bps,
+            redeemable_amount_under_management: credix_lp_depository_0
+                .redeemable_amount_under_management,
+            redeemable_amount_under_management_cap: credix_lp_depository_0
+                .redeemable_amount_under_management_cap,
+            mint_fn: Box::new(|collateral_amount| {
+                msg!(
+                    "[mint_with_router:mint_with_credix_lp_depository:{}]",
+                    collateral_amount
+                );
+                if collateral_amount > 0 {
+                    uxd_cpi::cpi::mint_with_credix_lp_depository(
+                        ctx.accounts.into_mint_with_credix_lp_depository_0_context(),
+                        collateral_amount,
+                    )?;
+                }
+                Ok(())
+            }),
+        },
+    ];
+
+    drop(controller);
+    drop(identity_depository);
+    drop(mercurial_vault_depository_0);
+    drop(credix_lp_depository_0);
+
+    // Compute the desired target amounts for each depository
+    let depositories_target_redeemable_amount = calculate_depositories_target_redeemable_amount(
+        maximum_after_mint_circulating_supply,
+        &depository_info
+            .iter()
+            .map(|depository_info| DepositoryInfoForTargetRedeemableAmount {
+                weight_bps: depository_info.weight_bps,
+                redeemable_amount_under_management_cap: depository_info
+                    .redeemable_amount_under_management_cap,
+            })
+            .collect::<Vec<DepositoryInfoForTargetRedeemableAmount>>(),
+    )?;
+
+    // Compute the desired mint amounts for each depository
+    let depositories_mint_collateral_amount = calculate_depositories_mint_collateral_amount(
+        collateral_amount,
+        &std::iter::zip(
+            depository_info.iter(),
+            depositories_target_redeemable_amount.iter(),
+        )
+        .map(|(depository_info, depository_target_redeemable_amount)| {
+            DepositoryInfoForMintCollateralAmount {
+                target_redeemable_amount: *depository_target_redeemable_amount,
+                redeemable_amount_under_management: depository_info
+                    .redeemable_amount_under_management,
+            }
+        })
+        .collect::<Vec<DepositoryInfoForMintCollateralAmount>>(),
+    )?;
+
+    // Call all the mint functions with the compute amounts
+    std::iter::zip(
+        depository_info.iter(),
+        depositories_mint_collateral_amount.iter(),
+    )
+    .try_for_each(|(depository_info, depository_mint_collateral_amount)| {
+        (depository_info.mint_fn)(*depository_mint_collateral_amount)
+    })?;
+
```

Comment:
Might possibly want to add some safe guard about the minted amount, matching the request or smthg? Not sure if superfluous

Reply:
Yes ser! already did in the prop test, that ensure that input == SUM(output)

over here: https://github.com/UXDProtocol/uxd-program/pull/251/files#diff-382aa24f0111d6ab6dcbb7737be6cae1c2e3c3031374e78949d17c3c546da450R254

I would say it is superfluous to do in the actual IX ? since its already prop-tested + checking for it would require complexifying the IX (accounting for precision loss and stuff)

What do you think ?